internal/app/synchronizer/evaluator_client.go

```diff
@@ -134,7 +150,77 @@ func (ec *evaluatorClient) grpcEvaluate(ctx context.Context, proposals []*pb.Mat
 	return results, nil
 }
 
-func (ec *evaluatorClient) httpEvaluate(ctx context.Context, proposals []*pb.Match) (results []*pb.Match, err error) {
-	// TODO: Implement a bidirectional HTTP streaming client in v0.7
-	return nil, nil
+func (ec *evaluatorClient) httpEvaluate(ctx context.Context, proposals []*pb.Match) ([]*pb.Match, error) {
+	reqr, reqw := io.Pipe()
+	proposalIDs := getMatchIds(proposals)
+	waitc := make(chan error)
+
+	go func() {
+		var m jsonpb.Marshaler
+		defer close(waitc)
```

Comment:
did you intend to close waitc here ? I see it being used in the main function too - wont there be a race?